File: sdk/client/java/openapi-test/src/test/java/opensdk/sdk/apis/klaytnDebug/others/DebugGetModifiedStorageNodesByNumberTest.java
package opensdk.sdk.apis.klaytnDebug.others;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertNull;

import java.io.IOException;
import opensdk.sdk.apis.constant.UrlConstants;
import opensdk.sdk.models.DebugGetModifiedStorageNodesByNumberResponse;
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.klaytn.OpenSDK;
@DisplayName("Debug RPC Test")
public class DebugGetModifiedStorageNodesByNumberTest {
  private final OpenSDK sdk = new OpenSDK(UrlConstants.SERVER_URL);
  @Disabled
  @Test
  @DisplayName("RPC debug_storageRangeAt")
  void whenRequestValid_ThenCall200ResponseReturns() throws IOException {
    String address = "0x73a7d19d14f7dfac5b799e405e22133b2adc57a6";
    Integer startBlockNum = 100;
    Integer endBlockNum = 200;
    DebugGetModifiedStorageNodesByNumberResponse response = sdk.debug.getModifiedStorageNodesByNumber(
        address, startBlockNum, endBlockNum).send();
    assertNotNull(response);
    assertNull(response.getError());
  }
}

File: web3rpc/sdk/client/javascript/openapi-test/test/debug/others/DebugGetModifiedStorageNodesByNumber.test.js
const OpenSdk = require("@klaytn/web3rpc");
const { expect } = require("@jest/globals");
const { RPC } = require("../../constant");

const sdk = new OpenSdk.DebugApi(new OpenSdk.ApiClient(RPC));

describe('debug_getModifiedStorageNodesByNumber.test API', () => {
    test.skip('should return debug_getModifiedStorageNodesByNumber', (done) => {

        let callbackOne = function (error, data, response) {
            expect(error).toBeNull();
            expect(data.result).toBeDefined()
            done();
        };

        const contractAddress = "0x73a7d19d14f7dfac5b799e405e22133b2adc57a6";
        const startBlockNum = 100;
        const endBlockNum = 200;

        sdk.getModifiedStorageNodesByNumber(contractAddress, startBlockNum, endBlockNum, {}, callbackOne);
    });
});

